File: app/src/main/java/com/handstandsam/shoppingapp/mockaccount/StubMappings.java
package com.handstandsam.shoppingapp.mockaccount;


import android.net.Uri;

import com.github.tomakehurst.wiremock.client.MappingBuilder;

import static com.github.tomakehurst.wiremock.client.WireMock.get;
import static com.github.tomakehurst.wiremock.client.WireMock.post;
import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;

public class StubMappings {


    public static MappingBuilder getCategories() {
        return get(urlEqualTo("/categories"));
    }

    public static MappingBuilder getItemsForCategory(String categoryId) {
        return get(urlEqualTo("/category/" + Uri.encode(categoryId) + "/items"));
    }

    public static MappingBuilder login() {
        return post(urlEqualTo("/login"));
    }

}

File: debug/src/main/java/com/handstandsam/shoppingapp/mockaccount/StubMappings.kt
package com.handstandsam.shoppingapp.mockaccount


import android.net.Uri
import com.github.tomakehurst.wiremock.client.MappingBuilder
import com.github.tomakehurst.wiremock.client.WireMock.*

object StubMappings {

    val categories: MappingBuilder
        @JvmStatic
        get() = get(urlEqualTo("/categories"))

    @JvmStatic
    fun getItemsForCategory(categoryId: String): MappingBuilder {
        return get(urlEqualTo("/category/" + Uri.encode(categoryId) + "/items"))
    }

    @JvmStatic
    fun login(): MappingBuilder {
        return post(urlEqualTo("/login"))
    }

}
